lock-impl/src/main/java/com/palantir/lock/logger/LockServiceStateLogger.java

```diff
@@ -141,10 +134,11 @@ private Multimap<ObfuscatedLockDescriptor, SimpleTokenInfo> generateHeldLocks(
                 .collect(ImmutableSetMultimap.toImmutableSetMultimap(Entry::getKey, Entry::getValue));
     }
 
-    private Map<ObfuscatedLockDescriptor, String> generateSyncState(
+    @Unsafe
+    private SetMultimap<@Safe ObfuscatedLockDescriptor, @Safe String> generateSyncState(
```

Comment:
can this return value be safe as well?

Reply:
This should actually be safe and it is flowing into a `SafeArg`, but something in the safety propagation is forcing this to be annotated as `@Unsafe`. I haven't had a chance to dig into what that is yet. @carterkozak may have ideas when he's back on if this is a limitation around propagating safety through `Multimap`/`SetMultimap`.

https://github.com/palantir/atlasdb/blob/ba7b17b5d19298b99261acc4cf68c02db7095f59/lock-impl/src/main/java/com/palantir/lock/logger/LogState.java#L55